src/slicer/SlicerArrows.js

```diff
@@ -0,0 +1,187 @@
+import {SLICING_GEOMETRY_COLOR} from '../constants';
+import CallbackProperty from 'cesium/Source/DataSources/CallbackProperty';
+import Entity from 'cesium/Source/DataSources/Entity';
+import Color from 'cesium/Source/Core/Color';
+import BoundingSphere from 'cesium/Source/Core/BoundingSphere';
+import Cartesian2 from 'cesium/Source/Core/Cartesian2';
+import ScreenSpaceEventType from 'cesium/Source/Core/ScreenSpaceEventType';
+import JulianDate from 'cesium/Source/Core/JulianDate';
+import ScreenSpaceEventHandler from 'cesium/Source/Core/ScreenSpaceEventHandler';
+import {getDirectionFromPoints} from '../cesiumutils';
+
+/**
+ * @typedef {object} ArrowListItem
+ * @property {string} side - arrow position label
+ * @property {string} uri - path to model
+ * @property {string} [oppositeSide - opposite arrow position label]
+ * @property {string} [position - arrow position. Required if 'positionUpdateCallback' is not provided]
+ * @property {string} [oppositePosition - position to create move axis. Required if no opposite arrow]
+ */
+
+export default class SlicerArrows {
+  /**
+   * Creates one or more entities and handle their move.
+   * @param {Viewer} viewer
+   * @param {DataSource} dataSource - dataSource to store entities
+   * @param {{
+   *          arrowsList: ArrowListItem[],
+   *          positionUpdateCallback: (function(string): Cartesian3),
+   *          moveCallback: (function(string, number, Cartesian3): void)
+   *        }} options
+   */
+  constructor(viewer, dataSource, options) {
+    this.viewer = viewer;
+    this.dataSource = dataSource;
+    this.moveCallback = options.moveCallback;
+    this.positionUpdateCallback = options.positionUpdateCallback;
+    this.arrowsList = options.arrowsList;
+    this.julianDate = new JulianDate();
+    this.eventHandler = null;
+    this.selectedArrow = null;
+  }
+
+  show() {
+    this.createMoveArrows();
+    this.eventHandler = new ScreenSpaceEventHandler(this.viewer.canvas);
+    this.eventHandler.setInputAction(this.onLeftDown.bind(this), ScreenSpaceEventType.LEFT_DOWN);
+    this.eventHandler.setInputAction(this.onMouseMove.bind(this), ScreenSpaceEventType.MOUSE_MOVE);
+    this.eventHandler.setInputAction(this.onLeftUp.bind(this), ScreenSpaceEventType.LEFT_UP);
+  }
+
+  hide() {
+    if (this.eventHandler) {
+      this.eventHandler.destroy();
+      this.eventHandler = null;
+    }
+  }
+
+  onLeftDown(event) {
+    const pickedObject = this.viewer.scene.pick(event.position);
+    const isModelPicked = pickedObject && pickedObject.id && pickedObject.id.model;
+    if (isModelPicked && pickedObject.id.properties && pickedObject.id.properties.side) {
+      this.selectedArrow = pickedObject.id;
+      this.viewer.scene.screenSpaceCameraController.enableInputs = false;
+    }
+  }
+
+  onLeftUp() {
+    if (this.selectedArrow) {
+      this.selectedArrow = null;
+      this.viewer.scene.screenSpaceCameraController.enableInputs = true;
```

Comment:
This looks ugly as the enableInputs might have been "false" when you did the leftDown.
I don't know how much we have code like this in the app. It would be nice to remember the previous value and restore it.